File: examples/dotnet/SendRcs.cs
using System;
using System.Threading.Tasks;
using Pushfy;

// Send an RCS rich card with SendRcsAsync.
// Only `to` and `text` are required; the rest turn it into a rich card with a button.
//   export PUSHFY_API_TOKEN=...
//   dotnet run
internal static class SendRcsExample
{
    public static async Task Main()
    {
        var token = Environment.GetEnvironmentVariable("PUSHFY_API_TOKEN");
        if (string.IsNullOrEmpty(token))
        {
            Console.Error.WriteLine("Set PUSHFY_API_TOKEN in your environment.");
            Environment.Exit(2);
        }

        using var pushfy = new PushfyClient(new PushfyClientOptions { ApiToken = token });

        try
        {
            var result = await pushfy.SendRcsAsync(
                to: "5511999999999",
                text: "Your order #1042 is on the way",
                title: "Order shipped",
                image: "https://cdn.example.com/box.jpg",
                url: "https://example.com/track/1042",
                cta: "Track order",
                extId: "rcs-order-1042");
            Console.WriteLine("Accepted: " + result);
        }
        catch (RateLimitException err)
        {
            Console.Error.WriteLine($"Rate limited. Retry after {err.RetryAfter}s.");
            Environment.Exit(1);
        }
        catch (InvalidRequestException err)
        {
            Console.Error.WriteLine($"Malformed RCS card: {err.Message}");
            Environment.Exit(1);
        }
        catch (ApiException err)
        {
            Console.Error.WriteLine($"Transient failure: {err.Status} {err.Message}");
            Environment.Exit(1);
        }
    }
}

File: examples/java/SendRcs.java
import com.pushfy.ApiException;
import com.pushfy.InvalidRequestException;
import com.pushfy.Pushfy;
import com.pushfy.RateLimitException;

/**
 * Send an RCS rich card with {@link Pushfy#sendRcs(Pushfy.Rcs)}.
 *
 * <p>Only {@code to} and {@code text} are required; title, image, url and cta
 * are optional and turn a plain message into a rich card with a tappable button.
 */
public class SendRcs {

    public static void main(String[] args) {
        String token = System.getenv("PUSHFY_API_TOKEN");
        if (token == null || token.isEmpty()) {
            System.err.println("Set PUSHFY_API_TOKEN in your environment.");
            System.exit(2);
        }

        Pushfy pushfy = Pushfy.builder()
                .apiToken(token)
                .build();

        Pushfy.Rcs card = new Pushfy.Rcs();
        card.to    = "5511999999999";
        card.title = "Order shipped";
        card.text  = "Your order #1042 is on the way";
        card.image = "https://cdn.example.com/box.jpg";
        card.url   = "https://example.com/track/1042";
        card.cta   = "Track order";
        card.extId = "rcs-order-1042";

        try {
            Object result = pushfy.sendRcs(card);
            System.out.println("Accepted: " + result);
        } catch (RateLimitException err) {
            System.err.println("Rate limited. Retry after " + err.retryAfter + "s.");
            System.exit(1);
        } catch (InvalidRequestException err) {
            System.err.println("Malformed RCS card: " + err.getMessage());
            System.exit(1);
        } catch (ApiException err) {
            System.err.println("Transient failure: " + err.status + " " + err.getMessage());
            System.exit(1);
        }
    }
}
